source/docs/software/vision-processing/apriltag/apriltag-intro.rst

```diff
@@ -0,0 +1,233 @@
+
+  .. code-tab:: java
+
+    // Coming soon!
+
+  .. code-tab:: c++
+
+    // Coming soon!
+
+In turn, using the `Pose3d` classes and the known positions of tags on the field, the robot's position on the field may be estimate.
+
+These estimates can be incorporated into the WPILib pose estimation classes.
+
+2D to 3D Ambiguity
+------------------
+
+The process of translating the four known corners of the target in the image (two-dimensional) into a real-world position relative to the camera (three-dimensional") is inherently ambiguous. That is to say, there are multiple real-world positions that result in the target corners ending up in the same spot in the image.
+
+Humans can often use lighting or background clues to understand how objects are oriented in space. However, computers do not have this benefit. They can be tricked by similar-looking targets:
+
+.. image:: images/planar_ambiguity1.png
+
+.. image:: images/planar_ambiguity2.png
+
+.. image:: images/planar_ambiguity3.png
+
+Resolving which position is "correct" can be done in a few different ways:
+
+1. Use your odometry history from all sensors to pick the pose closest to where you expect the robot to be.
+2. Reject poses which are very unlikely (ex: outside the field perimiter, or up in the air)
+3. Ignore pose estimates which are very close together (and hard to differentiate)
+4. Use multiple cameras to look at the same target, such that at least one camera can generate a good pose estimate
+5. Look at many targets at once, using each to generate multiple pose estimates. Discard the outlying estimates, use the ones which are tightly clustered together.
+
+Adjustable Parameters
+---------------------
+
+`Decimation factor` impacts how much the image is down-sampled before processing. Increasing it will increase detection speed, at the cost of not being able to see tags which are far away.
+
+`Blur` applies smoothing to the input image to decrease noise, which increases speed when fitting quads to pixels, at the cost of precision. For most good cameras, this may be left at zero.
+
+`Threads` changes the number of parallel threads which the algorithm uses to process the image. Certain steps may be sped up by allowing multithreading. In general, you want this to be approximately equal to the number of physical cores in your CPU, minus the number of cores which will be used for other processing tasks.
+
+Detailed information about the tunable parameters `can be found here <https://github.com/AprilRobotics/apriltag/wiki/AprilTag-User-Guide#tuning-the-detector-parameters>`_.
+
+Further Learning
+----------------
+
+The three major versions of AprilTags are described in three academic papers. It's recommended to read them in order, as each builds upon the previous:
+
+* :download:`AprilTags v1 <files/olson2011tags.pdf>`
+* :download:`AprilTags v2 <files/wang2016iros.pdf>`
+* :download:`AprilTags v3 <files/krogius2019iros.pdf>`
+* :download:`Pose Ambiguity <files/mmsp2014_spe.pdf>`
```

Comment:
Long term, I think we want to avoid hosting PDFs here on the GitHub. @AustinShalit can we get this placed somewhere?